pkg/provision/vultr.go

```diff
@@ -0,0 +1,217 @@
+package provision
+
+import (
+	"context"
+	"fmt"
+	"github.com/vultr/govultr"
+	"strconv"
+	"strings"
+)
+
+const vultrHostRunning = "ok"
+const exiteNodeTag = "inlets-exit-node"
+
+type VultrProvisioner struct {
+	client *govultr.Client
+}
+
+func NewVultrProvisioner(accessKey string) (*VultrProvisioner, error) {
+	return &VultrProvisioner{
+		client: govultr.NewClient(nil, accessKey),
+	}, nil
+}
+
+func (v *VultrProvisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
+
+	script, err := v.client.StartupScript.Create(context.Background(), host.Name, host.UserData, "boot")
+	if err != nil {
+		return nil, err
+	}
+
+	region, err := v.lookupRegion(host.Region)
+	if err != nil {
+		return nil, err
+	}
+
+	plan, err := strconv.Atoi(host.Plan)
+	if err != nil {
+		return nil, err
+	}
+
+	os, err := v.lookupOS(host.OS)
```

Comment:
Wondering why we don't just set the final OS id in create.go, instead of looking up during runtime. Is it because the final id is dynamic?

Reply:
you are right, the id is hardcoded in create.go, no need to do a lookup here